### PR TITLE
Move Prow to GHA

### PR DESCRIPTION
**Pull request by barchw**
**Description**

Changes proposed in this pull request:

- Move Goat jobs to GHA

**Related issues**
- #527

### CODE REVIEW

**Review by kolodziejczak — CHANGES_REQUESTED** [review]
We are missing AWS specific integration test on main and PR to release branches

**Comment by barchw** [comment]
> We are missing AWS specific integration test on main and PR to release branches

@kolodziejczak I would say we need to re-discuss that topic again
If anything we could still do it in follow up to this one, so that's how I would do it if we agree on running on PRs as well

File: hack/ci/gardener-integration-aws-specific.sh [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+      -n "garden-${GARDENER_PROJECT_NAME}"
+
+    exit
+}
+
+./hack/ci/provision-gardener.sh
+# Cleanup on exit, be it successful or on fail
+trap cleanup EXIT INT
+
+make install deploy aws-integration-test
```
Comment (kolodziejczak) [inline hack/ci/gardener-integration-aws-specific.sh:24]: Make target `aws-integration-test` already contains install and deploy embedded

File: hack/ci/gardener-integration-aws-specific.sh [hunk 2]
```diff
@@ -0,0 +1,24 @@
+#!/usr/bin/env bash
```
Comment (kolodziejczak) [inline hack/ci/gardener-integration-aws-specific.sh:1]: We have already a script for that in tests/integration/scripts. I assume this is a way of duplicating scripts so prow won't fail after merging?

File: .github/workflows/main-integration.yaml [hunk 3]
```diff
@@ -41,11 +41,118 @@ jobs:
...
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - uses: ./.github/actions/integration-test
+        with:
+          operator-image-name: "europe-docker.pkg.dev/kyma-project/prod/istio-manager:${{github.sha}}"
+          servers-memory: "16g"
+          agents: 2
+
+  istio-integration-gcp:
```
Comment (kolodziejczak) [inline .github/workflows/main-integration.yaml:58]: we didn't have this job on post main, but in my opinion more jobs is ok

File: .github/workflows/main-integration.yaml [hunk 4]
```diff
@@ -41,11 +41,118 @@ jobs:
...
+        with:
+          fetch-depth: 0
+      - run: ./tests/integration/scripts/gardener-kubeconfig.sh
+        shell: bash
+        env:
+          GARDENER_TOKEN: ${{ secrets.GARDENER_TOKEN }}
+      - uses: actions/setup-go@v4
+        with:
+          go-version: "1.21"
+      - run: make IMG="europe-docker.pkg.dev/kyma-project/prod/istio-manager:${{github.sha}}" gardener-istio-integration-test
```
Comment (kolodziejczak) [inline .github/workflows/main-integration.yaml:135]: This target also wasn't running on post main, but also in my opinion ok to have it here

File: .github/workflows/pull-integration-rel.yaml [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+          GARDENER_CLUSTER_VERSION: "1.27"
+          GARDENER_REGION: "eu-west-1"
+          GARDENER_ZONES: "eu-west-1b,eu-west-1c,eu-west-1a"
+          MACHINE_TYPE: "m5.xlarge"
+          DISK_SIZE: 50
+          DISK_TYPE: "gp2"
+          SCALER_MAX: 3
+          SCALER_MIN: 1
+
+  istio-integration-aws:
```
Comment (kolodziejczak) [inline .github/workflows/pull-integration-rel.yaml:103]: job with this target was running on GCP in prow

**Review by kolodziejczak — CHANGES_REQUESTED** [review]

**Review by kolodziejczak — APPROVED** [review]

**Pull request merged**